src/script/redmine-upkeep.py

```diff
@@ -0,0 +1,882 @@
+#!/usr/bin/python3
+
+# Copyright 2025 IBM, Inc.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# This script was generated with the assistance of an AI language model.
+#
+# This is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License version 2.1, as published by
+# the Free Software Foundation.  See file COPYING.
+
+import argparse
+import copy
+import itertools
+import json
+import logging
+import os
+import random
+import re
+import signal
+import sys
+
+from datetime import datetime, timedelta, timezone
+from getpass import getuser
+from os.path import expanduser
+
+import git # https://github.com/gitpython-developers/gitpython
+import redminelib # https://pypi.org/project/python-redmine/
+import requests
+
+GITHUB_TOKEN = None
+try:
+    with open(expanduser("~/.github_token")) as f:
+        GITHUB_TOKEN = f.read().strip()
+except FileNotFoundError:
+    pass
+GITHUB_TOKEN = os.getenv("GITHUB_TOKEN", GITHUB_TOKEN)
+
+GITHUB_USER = os.getenv("GITHUB_USER", os.getenv("GITHUB_USER", getuser()))
+GITHUB_ORG = "ceph"
+GITHUB_REPO = "ceph"
+GITHUB_API_ENDPOINT = f"https://api.github.com/repos/{GITHUB_ORG}/{GITHUB_REPO}"
+
+REDMINE_CUSTOM_FIELD_ID_BACKPORT = 2
+REDMINE_CUSTOM_FIELD_ID_RELEASE = 16
+REDMINE_CUSTOM_FIELD_ID_PULL_REQUEST_ID = 21
+REDMINE_CUSTOM_FIELD_ID_TAGS = 31
+REDMINE_CUSTOM_FIELD_ID_MERGE_COMMIT = 33
+REDMINE_CUSTOM_FIELD_ID_FIXED_IN = 34
+REDMINE_CUSTOM_FIELD_ID_RELEASED_IN = 35
+REDMINE_CUSTOM_FIELD_ID_UPKEEP_TIMESTAMP = 37
+
+REDMINE_STATUS_ID_RESOLVED = 3
+REDMINE_STATUS_ID_REJECTED = 6
+REDMINE_STATUS_ID_FIX_UNDER_REVIEW = 13
+REDMINE_STATUS_ID_PENDING_BACKPORT = 14
+
+REDMINE_TRACKER_ID_BACKPORT = 9
+
+REDMINE_STATUS_ID_PENDING_BACKPORT = 14
+REDMINE_STATUS_ID_RESOLVED = 3
+
+REDMINE_ENDPOINT = "https://tracker.ceph.com"
+REDMINE_API_KEY = None
+try:
+    with open(expanduser("~/.redmine_key")) as f:
+        REDMINE_API_KEY = f.read().strip()
+except FileNotFoundError:
+    pass
+REDMINE_API_KEY = os.getenv("REDMINE_API_KEY", REDMINE_API_KEY)
+
+# Global flag for GitHub Actions output format
+IS_GITHUB_ACTION = False
```

Comment:
[Reference](https://docs.github.com/en/actions/reference/variables-reference):
```suggestion
IS_GITHUB_ACTION = os.getenv("GITHUB_ACTION")
```